lib/cockpit-components-file-autocomplete.jsx

```diff
@@ -1,202 +0,0 @@
-            this.updateFiles(value[0].path);
-        }
-    }
-
-    onInputChanged(value) {
-        if (this.props.onChange)
-            this.props.onChange(value);
-
-        this.setState({ value });
-
-        this.debouncedChange(value);
-    }
-
-    render() {
-        const placeholder = this.props.placeholder || _("Path to file");
-
-        return (
-            <FormGroup validationState={this.state.error && !this.state.hasFocus ? 'error' : undefined}>
-                <TypeAheadSelect
-                    id={this.props.id}
-                    labelKey='path'
-                    placeholder={placeholder}
-                    paginate={false}
-                    onChange={this.onValueChanged}
-                    onInputChange={this.onInputChanged}
-                    options={this.state.displayFiles}
-                    onKeyDown={ev => { // Capture ESC event
-                        if (ev.keyCode == 27) {
-                            ev.persist();
-                            ev.nativeEvent.stopImmediatePropagation();
-                            ev.stopPropagation();
-                        }
-                    }}
-                    renderMenu={(results, menuProps) => {
-                        // Hide the menu when there are no results.
-                        if (!results.length) {
-                            return null;
-                        }
-                        return <TypeAheadSelect.TypeaheadMenu {...menuProps} labelKey='path' options={results} />;
-                    }}
-                    onFocus={() => this.setState({ hasFocus: true, error: undefined })}
-                    onBlur={() => {
-                        let value = this.state.value;
-
-                        if (value.lastIndexOf('/') == value.length - 1)
-                            value = value.slice(0, value.length - 1);
-
-                        if (this.props.fileExists && value && (value + '/') != this.state.directory &&
-                            !this.state.displayFiles.find(file => (file.type == 'directory' ? (this.state.value + '/') : this.state.value) == file.path)) {
-                            if (!this.state.error)
-                                this.setState({ error: cockpit.format("No such file or directory '$0'", this.state.value) });
-                        }
-                        this.setState({ hasFocus: false });
-                    }}
-                    open={this.state.hasFocus}
-                />
-                { this.state.error && !this.state.hasFocus &&
-                <HelpBlock>
-                    <p className="text-danger">{this.state.error}</p>
-                </HelpBlock> }
-            </FormGroup>
-        );
-    }
-}
-
-FileAutoComplete.defaultProps = {
-    fileExists: true,
-};
-
-FileAutoComplete.propTypes = {
-    id: PropTypes.string,
-    placeholder: PropTypes.string,
-    superuser: PropTypes.string,
-    onChange: PropTypes.func,
-    fileExists: PropTypes.bool,
-};
```

Comment:
Oh, maybe I should have left a comment about that somewhere, that this component was customized, so it would accept `fileExists` argument, which would make it accept a path to a non-existent file.
It's useful because cockpit-certificates allows you to also input a file to a non-existent certificate key file. In that case it generate the key for you.
If user inputs a path to an existing file, that file is used as a key.

Reply:
Right, I posted a PR and now the original component also support such a property https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/cockpit-components-file-autocomplete.jsx#L197

Reply:
pkg/lib is just a read only directory, we should copy it over from master and never change it.